rebasehelper/application.py

```diff
@@ -482,73 +481,113 @@ def _prepare_rebased_repository(cls, patches, rebased_sources_dir):
         repo.index.commit('Initial commit', skip_hooks=True)
         return repo
 
-    def build_packages(self):
+    @staticmethod
+    def _sanitize_build_dict(build_dict):
+        blacklist = [
+            'builds_nowait',
+            'build_tasks',
+            'builder_options',
+            'srpm_builder_options',
+        ]
+        return {k: v for k, v in six.iteritems(build_dict) if k not in blacklist}
+
+    def build_source_packages(self):
+        try:
+            builder = SRPMBuilder(self.conf.srpm_buildtool)
+        except NotImplementedError as e:
+            raise RebaseHelperError('{}. Supported SRPM build tools are {}'.format(
+                six.text_type(e), SRPMBuilder.get_supported_tools()))
+
+        for version in ['old', 'new']:
+            results_dir = '{}-build'.format(os.path.join(self.results_dir, version))
+            spec = self.spec_file if version == 'old' else self.rebase_spec_file
+            package_name = spec.get_package_name()
+            package_version = spec.get_version()
+            package_full_version = spec.get_full_version()
+            logger.info('Building source package for %s version %s', package_name, package_full_version)
+            build_dict = dict(
+                name=package_name,
+                version=package_version,
+                srpm_buildtool=self.conf.srpm_buildtool,
+                srpm_builder_options=self.conf.srpm_builder_options)
+            try:
+                build_dict.update(builder.build(spec, results_dir, **build_dict))
+                build_dict = self._sanitize_build_dict(build_dict)
+                results_store.set_build_data(version, build_dict)
+            except RebaseHelperError:
+                raise
+            except SourcePackageBuildError as e:
+                build_dict.update(builder.get_logs())
+                build_dict['source_package_build_error'] = six.text_type(e)
```

Comment:
If I am not wrong, we are not storing errors from binary builds in this way. Can you explain?

Reply:
Actually I haven't touched this, just moved it from the original `build_packages()` method.
It certainly is possible that it's useless. But I plan to rework all the `ResultsStore` related things anyway.